src/main/java/com/griddynamics/cd/nrp/internal/model/config/ReplicationPluginConfigurationStorage.java

```diff
@@ -0,0 +1,135 @@
+package com.griddynamics.cd.nrp.internal.model.config;
+
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Named(value = "replicationPluginConfigurationStorage")
+@Singleton
+public class ReplicationPluginConfigurationStorage {
+
+    Logger logger =
+            LoggerFactory.getLogger(ReplicationPluginConfigurationStorage.class);
+
+    private final AtomicReference<String> masterServerURLPrefix = new AtomicReference<>();
```

Comment:
What's the point for `AtomicReference`? From the code it seems that `volatile` should be enough.